src/Tasks/Hash.cs

```diff
@@ -37,61 +35,120 @@ public class Hash : TaskExtension
         [Output]
         public string HashResult { get; set; }
 
+        private static readonly char ItemSeparatorCharacter = '\u2028';
+
+        private static readonly Encoding encoding = Encoding.UTF8;
+
+        private static readonly byte[] ItemSeparatorCharacterBytes = encoding.GetBytes(new char[] { ItemSeparatorCharacter });
+
+        private const int sha1BufferSize = 512;
+
+        private const int maxInputChunkLength = 256;
+
         /// <summary>
         /// Execute the task.
         /// </summary>
         public override bool Execute()
         {
-            if (ItemsToHash?.Length > 0)
+            if (ItemsToHash != null && ItemsToHash.Length > 0)
             {
                 using (var sha1 = SHA1.Create())
                 {
-                    var concatenatedItemStringSize = ComputeStringSize(ItemsToHash);
+                    // Buffer in which bytes of the strings are to be stored until their number reachs the limit size.
+                    // Once the limit is reached, the sha1.TransformBlock is be run on all the bytes of this buffer.
+                    byte[] sha1Buffer = System.Buffers.ArrayPool<byte>.Shared.Rent(sha1BufferSize);
 
-                    var hashStringSize = sha1.HashSize;
+                    int maxItemStringSize = encoding.GetMaxByteCount(Math.Min(ComputeMaxItemSpecLength(ItemsToHash), maxInputChunkLength));
+                    byte[] bytesBuffer = System.Buffers.ArrayPool<byte>.Shared.Rent(maxItemStringSize);
```

Comment:
Here I am not completely sure that it is a good idea to rent a buffer from a pool. 
We might just use allocations instead here and it will not be much worse.
What do you think of using `System.Buffers` in a task?  Could it lead to any problems? For example, for users who would like to use this task somehow?

Reply:
They'd have to JIT something extra if they hadn't already...it looks like we use System.Buffers in InterningBinaryReader.cs and System.Buffers.Binary in NodeProviderOutOfProcBase.cs. I'm not sure if System.Buffers.Binary is a separate assembly, but if it is, maybe not?

Reply:
If System.Buffers is present (net6+), the BCL already uses it in many places, so I guess it's JITted already